client/src/features/editors/list/EventList.jsx

```diff
@@ -19,21 +20,24 @@ export default function EventList(props) {
   const [selectedId, setSelectedId] = useState(null);
   const [nextId, setNextId] = useState(null);
   const cursorRef = createRef();
-  const { showQuickEntry } = useContext(LocalEventSettingsContext);
-
-  const insertAtCursor = useCallback((type, cursor) => {
-    if (cursor === -1) {
-      eventsHandler('add', { type: type });
-    } else {
-      const previousEvent = events[cursor];
-      const nextEvent = events[cursor + 1];
-      if (type === 'event') {
-        eventsHandler('add', { type: type, after: previousEvent.id });
-      } else if (previousEvent?.type !== type && nextEvent?.type !== type) {
-        eventsHandler('add', { type: type, after: previousEvent.id });
+  const showQuickEntry = useAtomValue(showQuickEntryAtom);
+
+  const insertAtCursor = useCallback(
```

Comment:
I know this is not part of the PR, but it seems we don't need `useCallback` in this component.

Reply:
Removing the callback here would mean changing the code structure so not to break the next callback along `handleKeyPress`
This component will come under scrutiny in #55 and I will revise then